internal/metastore/kv/datacoord/kv_catalog.go

```diff
@@ -172,6 +172,47 @@ func (kc *Catalog) AlterSegmentsAndAddNewSegment(ctx context.Context, segments [
 	return nil
 }
 
+// RevertAlterSegmentsAndAddNewSegment reverts the metastore operation of AtlerSegmentsAndAddNewSegment
+func (kc *Catalog) RevertAlterSegmentsAndAddNewSegment(ctx context.Context, oldSegments []*datapb.SegmentInfo, removeSegment *datapb.SegmentInfo) error {
+	var (
+		data     = make(map[string]string)
+		removals []string
+	)
+
+	for _, s := range oldSegments {
+		k, v, err := buildSegmentKeyValuePair(s)
+		if err != nil {
+			return err
+		}
+		data[k] = v
+	}
+
+	if removeSegment.NumOfRows > 0 {
```

Comment:
Why do we check `NumOfRows` before removing it?

Reply:
Because if the segment's NumOfRows == 0, it won't be stored, so don't need to remove it in revert